### PR TITLE
Install Docker, Docker Compose and Pyenv alongside linux-workstation

### PR DESCRIPTION
**Pull request by ciaransweet**

### CODE REVIEW

File: ansible/roles/docker/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    name: ['yum-utils', 'device-mapper-persistent-data', 'lvm2']
+    state: present
+
+- name: Docker YUM repo present
+  tags:
+   - docker
+  yum_repository:
+    name: docker-ce
+    description: Docker-ce YUM repo
+    baseurl: https://download.docker.com/linux/centos/7/x86_64/stable
```
Comment (ghost) [inline ansible/roles/docker/tasks/main.yml:16]: Add following line:
gpgkey: https://download.docker.com/linux/centos/gpg

Following step currently fails from fresh install with error: TASK [docker : Docker present] ***************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "changes": {"installed": ["docker-ce"]}, "msg": "warning: /var/cache/yum/x86_64/7/docker-ce/packages/docker-ce-18.09.7-3.el7.x86_64.rpm: Header V4 RSA/SHA512 Signature, key ID 621e9f35: NOKEY\n\n\nPublic key for docker-ce-18.09.7-3.el7.x86_64.rpm is not installed\n"

File: ansible/roles/docker/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  service:
+    name: docker
+    state: started
+
+- name: User in Docker group
+  tags:
+    - docker
+  user:
+    name: root
+    groups: docker
```
Comment (ghost) [inline ansible/roles/docker/tasks/main.yml:38]: Add local default user to docker group as well as root user.

File: ansible/roles/pyenv/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
+
+- name: pyenv depencencies present
+  yum:
+    name: ['gcc', 'zlib-devel', 'bzip2', 'bzip2-devel', 'readline-devel', 'sqlite', 'sqlite-devel',
+    'openssl-devel', 'tk-devel', 'libffi-devel']
+    state: present
```
Comment (ghost) [inline ansible/roles/pyenv/tasks/main.yml:7]: [picky] alphabetical order

File: ansible/roles/docker/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+
+- name: Docker YUM depencies present
+  tags:
+    - docker
+  yum:
+    name: ['yum-utils', 'device-mapper-persistent-data', 'lvm2']
```
Comment (ghost) [inline ansible/roles/docker/tasks/main.yml]: [picky] alphabetical order

File: ansible/roles/pyenv/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    line: 'if command -v pyenv 1>/dev/null 2>&1; then eval "$(pyenv init -)";fi'
+    insertafter: EOF
+
+- name: pyenv virtualenv present
+  tags:
+    - pyenv
+  git:
+    repo: https://github.com/pyenv/pyenv-virtualenv.git
+    dest: '{{ pyenv_root }}/plugins/pyenv-virtualenv'
+    version: 294f64f76b6b7fbf1a22a4ebba7710faa75c21f7
```
Comment (ghost) [inline ansible/roles/pyenv/tasks/main.yml]: Make this an ansible variable and put in the defaults.yml file for the role.

File: ansible/roles/pyenv/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    state: directory
+    mode: '0777'
+
+- name: pyenv source cloned
+  tags:
+      - pyenv
+  git:
+    repo: https://github.com/pyenv/pyenv.git
+    dest: '{{ pyenv_root }}'
+    version: eb68ec9488f0df1f668e9272dd5bd8854edf1dff
```
Comment (ghost) [inline ansible/roles/pyenv/tasks/main.yml]: Make this an ansible variable and put in the defaults.yml file for the role.

**Review by ghost — CHANGES_REQUESTED** [review]

File: ansible/roles/docker/tasks/main.yml (hunk 4 above)
Comment (ciaransweet) [reply ansible/roles/docker/tasks/main.yml]: Done

File: ansible/roles/docker/tasks/main.yml (hunk 2 above)
Comment (ciaransweet) [reply ansible/roles/docker/tasks/main.yml:38]: Done

File: ansible/roles/pyenv/tasks/main.yml (hunk 3 above)
Comment (ciaransweet) [reply ansible/roles/pyenv/tasks/main.yml:7]: Done

File: ansible/roles/pyenv/tasks/main.yml (hunk 6 above)
Comment (ciaransweet) [reply ansible/roles/pyenv/tasks/main.yml]: Done

File: ansible/roles/pyenv/tasks/main.yml (hunk 5 above)
Comment (ciaransweet) [reply ansible/roles/pyenv/tasks/main.yml]: Done

File: ansible/roles/docker/tasks/main.yml (hunk 1 above)
Comment (ciaransweet) [reply ansible/roles/docker/tasks/main.yml:16]: Done

**Review by ghost — APPROVED** [review]

**Pull request merged**